src/workerd/api/sql-test.js

```diff
@@ -165,6 +180,148 @@ function test(sql) {
+    assert.equal(results[0].id, 2) // Doc 2 comes first (sorted by rank)
+    assert.equal(results[1].id, 1)
+  }
+  // highlight() function
+  {
+    let results = Array.from(sql.exec(`
+        SELECT highlight(documents_fts, 2, '<b>', '</b>') as output FROM documents_fts WHERE documents_fts MATCH '2' ORDER BY rank;
+    `));
+    assert.equal(results.length, 2)
+    assert.equal(results[0].output, `This is the content of document <b>2</b> (of <b>2</b>).`) // two matches, two highlights
+    assert.equal(results[1].output, `This is the contents of document 1 (of <b>2</b>).`)
+  }
+  // snippet() function
+  {
+    let results = Array.from(sql.exec(`
+        SELECT snippet(documents_fts, 2, '<b>', '</b>', '...', 4) as output FROM documents_fts WHERE documents_fts MATCH '2' ORDER BY rank;
+    `));
+    assert.equal(results.length, 2)
+    assert.equal(results[0].output, `...document <b>2</b> (of <b>2</b>).`) // two matches, two highlights
+    assert.equal(results[1].output, `...document 1 (of <b>2</b>).`)
+  }
+
+  // Complex queries
+
+  // List table info
+  {
+    let result = [...sql.exec(`
+        SELECT name as tbl_name,
+               ncol as num_columns
+        FROM pragma_table_list
+        WHERE TYPE = "table"
+          AND tbl_name NOT LIKE "sqlite_%"
+          AND tbl_name NOT LIKE "d1_%"
+          AND tbl_name NOT LIKE "_cf_%"`)];
+    assert.equal(result.length, 2);
+    assert.equal(result[0].tbl_name, 'myTable');
+    assert.equal(result[0].num_columns, 2);
+    assert.equal(result[1].tbl_name, 'documents');
+    assert.equal(result[1].num_columns, 3);
+  }
+
+  // Let the current open transaction commit. We have to do this before playing with the
+  // foreign_keys pragma because it doesn't work while a transaction is open.
+  await scheduler.wait(1);
+
+  let assertValidBool = (name, val) => {
+    sql.exec("PRAGMA foreign_keys = " + name + ";");
+    assert.equal([...sql.exec("PRAGMA foreign_keys;")][0].foreign_keys, val);
+  };
+  let assertInvalidBool = (name, msg) => {
+    requireException(() => sql.exec("PRAGMA foreign_keys = " + name + ";"),
+        msg || "not authorized");
+  };
+
+  assertValidBool("true", 1);
+  assertValidBool("false", 0);
+  assertValidBool("on", 1);
+  assertValidBool("off", 0);
+  assertValidBool("yes", 1);
+  assertValidBool("no", 0);
+  assertValidBool("1", 1);
+  assertValidBool("0", 0);
+
+  // case-insensitive
+  assertValidBool("tRuE", 1);
+  assertValidBool("NO", 0);
+
+  // quoted
+  assertValidBool("'true'", 1);
+  assertValidBool("\"yes\"", 1);
+  assertValidBool("\"0\"", 0);
+
+  // whitespace is trimmed by sqlite before passing to authorizer
+  assertValidBool("  true    ", 1);
+
+  // Don't accept anything invalid...
+  assertInvalidBool("abcd");
+  assertInvalidBool("\"foo\"");
+  assertInvalidBool("'yes", "unrecognized token");
+
+  // Test database size interface.
```

Comment:
nb: should we add a test to make sure that the limit does in fact actually limit...